File: app/src/main/java/com/jay/currencyconverter/di/NetworkComponent.java
package com.jay.currencyconverter.di;

import com.jay.currencyconverter.model.currencyExchange.bank.Banks;
import com.jay.currencyconverter.model.currencyExchange.nbu.Nbu;

import java.util.List;

import dagger.Component;
import io.reactivex.Observable;

@Component(modules = {NetworkModule.class})
public interface NetworkComponent {

    Observable<Banks> getBankExchange();

    Observable<List<Nbu>> getNbuExchange();
}

File: app/src/main/java/com/jay/currencyconverter/di/NetworkComponent.kt
package com.jay.currencyconverter.di

import com.jay.currencyconverter.annotation.BankExchange
import com.jay.currencyconverter.annotation.NbuExchange
import com.jay.currencyconverter.annotation.NbuExchangeByDateAndCurrency
import com.jay.currencyconverter.api.CurrencyExchangeRateApi
import dagger.Component

@Component(modules = [NetworkModule::class])
interface NetworkComponent {

    @get:BankExchange
    val bankExchange: CurrencyExchangeRateApi

    @get:NbuExchange
    val nbuExchange: CurrencyExchangeRateApi

    @get:NbuExchangeByDateAndCurrency
    val nbuExchangeByDateAndCurrency: CurrencyExchangeRateApi
}